src/Formidable.re

```diff
@@ -1,636 +0,0 @@
-open FormidableExtra;
-
-module Events = FormidableEvents;
-module Validations = FormidableValidations;
-
-module type Type = {type t;};
-
-module type Values = {
-  include Type;
-
-  let init: t;
-};
-
-module type Handlers = {
-  type values;
-
-  type error;
-
-  let onSubmit: option(values => unit);
-
-  let onSubmitError: option((values, list(error)) => unit);
-};
-
-module States = {
-  module Field = {
-    module Status = {
-      type t('error) = [
-        | `pristine
-        | `valid
-        | `touched
-        | `errors(list('error))
-      ];
-    };
-
-    [@bs.deriving accessors]
-    type t('values, 'error) = {
-      status: Status.t('error),
-      validate:
-        (option(Validations.Strategy.t), 'values) => Status.t('error),
-    };
-
-    let isPristine =
-      fun
-      | {status: `pristine} => true
-      | _ => false;
-
-    let isValid =
-      fun
-      | {status: `valid} => true
-      | _ => false;
-
-    let isTouched =
-      fun
-      | {status: `touched} => true
-      | _ => false;
-
-    let hasErrors =
-      fun
-      | {status: `errors(_)} => true
-      | _ => false;
-
-    let getErrors =
-      fun
-      | {status: `errors(errors)} => Some(errors)
-      | _ => None;
-
-    let reset = field => {...field, status: `pristine};
-  };
-
-  let isPristine = fields => fields->Map.String.every(_ => Field.isPristine);
-
-  let isValid = fields => fields->Map.String.every(_ => Field.isValid);
-
-  let isTouched = fields => !fields->Map.String.some(_ => Field.isPristine);
```

Comment:
not sure why here is !fields. Was it a mistake? (I made it as fields in rescript file)

Reply:
You're right, I think it should be `let isTouched = fields => fields->Map.String.some(_ => !Field.isPristine)` in ReScript